SU-SWS/stanford_basic
# D8CORE-1472: Updates to lockup settings for logo support

# READY FOR REVIEW

# Summary
- Adding lockup options and the ability to upload a custom logo
- Allows lockup to have only a logo
- Updates lockup form to hide/show fields that are available on each option.

# Review By (Date)
- EOD Friday the 24th if you can.

# Urgency
- Medium / High

# Steps to Test

1. Do a fresh build of the 1.x branch of the acsf-cardinalsites stack with the stanford_profile
2. Check out this branch
3. Clear caches
4. Go to /admin/appearance/settings/stanford_basic
5. Select different lockup options from the drop down and see the available line input fields hide/show corresponding to the option selected.
6. Select an option other than `Logo Only` and fill out the line fields.
7. Save form and review on the home page you see what you would have expected to see.
8. Repeat with another option other than logo only.
9. Select `Logo Only` and save the form.
10. Review the home page and only see the Stanford wordmark.
11. Go back to the /admin/appearance/settings/stanford_basic and uncheck `Use the logo supplied by the theme` so you can upload an image. 
12. Upload a large image.
13. Save the form and review the home page.
14. See only the uploaded logo and at a reasonable size.
15. Go back to /admin/appearance/settings/stanford_basic and select `Option A` from the list of options and fill out the lines.
16. Save the form and go to the home page to see your uploaded logo and the lockup text. 

# Affected Projects or Products
- ACSF Stacks
- SAA/OOD
- SOE
- stanford_profile
- Earth

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/D8CORE-1472
- https://stanfordits.atlassian.net/browse/D8CORE-1768
- https://stanfordits.atlassian.net/browse/D8CORE-1771
- https://github.com/SU-SWS/stanford_profile/pull/220
- @rmundstock 
- @buttonwillowsix 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
